### PR TITLE
Add SignatureData

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
Add a new type to contain the members of the Quote3 signature data,
`sgx_ql_ecdsa_sig_data_t`.

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #200** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/200" target="_blank">Graphite</a> 
    * **PR #202** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/202" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/202?utm_source=stack-comment).

File: dcap/types/src/quote3.rs [hunk 1]
```diff
@@ -19,13 +23,11 @@ use subtle::ConstantTimeEq;
...
-// certification data.
-const QUOTE_SIZE: usize =
-    mem::size_of::<sgx_quote3_t>() + mem::size_of::<sgx_ql_ecdsa_sig_data_t>();
+// Size of the Key
+const KEY_SIZE: usize = 64;
 
-// The offset to the authentication data
-const AUTH_DATA_OFFSET: usize = QUOTE_SIZE;
+// Size of a Signature
+const SIGNATURE_SIZE: usize = 64;
```
Comment (nick-mobilecoin) [inline dcap/types/src/quote3.rs:30]: This and the `MIN_SIGNATURE_DATA_SIZE` are pretty similar up for ideas on better names, they are local constants so it probably isn't super critical name wise

File: dcap/types/src/quote3.rs [hunk 2]
```diff
@@ -73,6 +77,13 @@ impl Error {
     }
 }
 
+impl From<ecdsa::Error> for Error {
+    fn from(_: ecdsa::Error) -> Self {
+        // ecdsa errors are opaque to avoid side channel leakage
```
Comment (jcape) [inline dcap/types/src/quote3.rs:82]:
```suggestion
        // ecdsa::Error is opaque, and only provides additional information via `std::Error` impl.
```

**Review by jcape — APPROVED** [review]

**Review by awygle — APPROVED** [review]
This looks good to me. I appreciate all the tests.

**Pull request merged**